test/OpenTelemetry.Instrumentation.Cassandra.Tests/CassandraInstrumentationTests.cs

```diff
@@ -0,0 +1,151 @@
+// <copyright file="CassandraInstrumentationTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Cassandra;
+using Cassandra.Mapping;
+using Cassandra.Metrics;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Tests;
+using Xunit;
+using CassandraData = Cassandra.Data.Linq;
+
+namespace OpenTelemetry.Instrumentation.Cassandra.Tests;
+
+[Collection("Cassandra")]
+public class CassandraInstrumentationTests
+{
+    private const int MaxTimeToAllowForFlush = 20000;
+
+    private const string CassandraConnectionStringEnvName = "OTEL_CASSANDRA_CONNECTION_STRING";
+    private static readonly string? CassandraConnectionString = SkipUnlessEnvVarFoundFactAttribute.GetEnvironmentVariable(CassandraConnectionStringEnvName);
+
+    [Trait("CategoryName", "CassandraIntegrationTests")]
+    [SkipUnlessEnvVarFoundFact(CassandraConnectionStringEnvName)]
```

Comment:
So I should send a PR to https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation to add the integration tests right? if Yes, I can do it as follow up.

Reply:
You can consider create integration tests in this repository.
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation repository is used to automatic instrumentation.
If you would like to include it for this scenario we probably can do this, but it will be tricky - as I understand CassandraBuilderExtensions.WithOpenTelemetryMetrics has to be called for each instance.

Reply:
Thanks, I'll follow up